### PR TITLE
Remove unnecessary code from image.transform.affine_transform

### PR DESCRIPTION
**Pull request by larrymanley**

### CODE REVIEW

**Comment by larrymanley** [comment]
ping @Cadair I've checked the documentation of previous versions (back to 0.3) of skimage and I see no reference to warp needing 0:1 normalization

File: sunpy/image/transform.py [hunk 1]
```diff
@@ -130,20 +128,7 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
                           RuntimeWarning)
             adjusted_image = np.nan_to_num(adjusted_image)
```
Comment (wafels) [inline sunpy/image/transform.py]: I think in general fixing input for users is to be avoided.  This is because we can't really anticipate what the user intends.  Pixels that are NaNs or infs can be replaced in a multitude of different ways.

Comment (larrymanley) [inline sunpy/image/transform.py]: I can remove that code also; I don't see any reference to NaNs not being allowed for higher-order interpolation either

Comment (Cadair) [inline sunpy/image/transform.py]: NaNs are casted to zeros in scipy and for higher order in scikit image, this function doesn't do it.

**Comment by Cadair** [comment]
some not very helpful info here: http://scikit-image.org/docs/stable/user_guide/data_types.html#input-types

**Comment by larrymanley** [comment]
Should I remove lines 128-131 also?

**Comment by nabobalis** [comment]
What lines are those?

**Comment by larrymanley** [comment]
`         if np.any(np.isnan(adjusted_image)) and order >= 4:
             warnings.warn("Setting NaNs to 0 for higher-order scikit-image rotation",
                            RuntimeWarning)
              adjusted_image = np.nan_to_num(adjusted_image)
`

**Comment by nabobalis** [comment]
There was some debate on whether the code could be removed. No one seemed to be sure. I think there will be a larger discussion down the line. 

For now, remove it and could you merge/rebase master into the PR? We have had to update things to get Travis to behave recently.

**Comment by nabobalis** [comment]
So it looks like the changes are causing one test to fail. 

```python
  def test_iris_rot():
        test_data = os.path.join(sunpy.data.test.rootdir,'iris_l2_20130801_074720_4040000014_SJI_1400_t000.fits')
        iris_cube = iris.SJI_to_cube(test_data, start=0, stop=None, hdu=0)
        irismap = iris_cube.maps[0]
>       irismap_rot = irismap.rotate()
sunpy/instr/tests/test_iris.py:25: 
```
with error
```python
ValueError: Images of type float must be between -1 and 1.
skimage/util/dtype.py:236: ValueError
```

So I think the code we removed is fine since it doesn't seem to fail any of the other tests for this code. There might be an issue for irismap.rotate().

**Comment by Cadair** [comment]
This error on first glace seems to indicate the code was there for a reason. More investigation is required.

**Comment by larrymanley** [comment]
If the data type is `float64` it doesn't need to be in the [-1, 1] range.  The IRIS data is `float32`.  Should we just cast everything to `float64`?

**Comment by Cadair** [comment]
you what, really?! why?! I mean cast to `float64` is not unreasonable...

**Comment by nabobalis** [comment]
Looks like this change fixed the IRIS test but broke another test.

```python
    def test_nan_skimage_high(identity):
        # Test replacement of NaN values for scikit-image rotation with order >=4
        in_arr = np.array([[np.nan]])
        out_arr = affine_transform(in_arr, rmatrix=identity, order=4)
>       assert not np.all(np.isnan(out_arr))
E       AssertionError: assert not True
E        +  where True = <function all at 0x112041488>(array([[ True]], dtype=bool))
E        +    where <function all at 0x112041488> = np.all
E        +    and   array([[ True]], dtype=bool) = <ufunc 'isnan'>(array([[ nan]]))
E        +      where <ufunc 'isnan'> = np.isnan
sunpy/image/tests/test_transform.py:209: AssertionError
```

I think we need more tests for edge cases and input variety before we should change this code. I'm not sure we know enough.

**Comment by Cadair** [comment]
Well that test looks like it was designed to test the section of the code that was removed, so I am not surprised it is failing. Could someone test the `GenericMap.rotate()` method, with `order=4` (default I think) using HMI LOS_magnetogram data to test that not manually removing the NaNs doesn't break things?

**Comment by Cadair** [comment]
and for bonus points, make it into a figure test.

**Comment by larrymanley** [comment]
HMI LOS data gets scaled to [-0.1, 0.1].  I tried passing `preserve_range=True` to `warp` but that didn't help.  The skimage docs state "If the input image has a float type, intensity values are not modified and can be outside the ranges [0.0, 1.0] or [-1.0, 1.0]."  Should I open an issue on scikit-image regarding the treatment of NaNs, or can we set NaNs to zero as with scipy?

**Comment by larrymanley** [comment]
My apologies for the confusion.  Why did I remove the nan_to_num code?

**Comment by larrymanley** [comment]
ping @cadair
